File: internal/content/testdata/bench/rust/perfect-square-check.rs
fn is_perfect_square(num: i64) -> bool {
    if num < 1 {
        return false;
    }
    let mut lo: i64 = 1;
    let mut hi: i64 = num;
    while lo <= hi {
        let mid = lo + (hi - lo) / 2;
        // Compare mid*mid to num without overflow.
        if mid > num / mid {
            hi = mid - 1;
        } else if mid * mid == num {
            return true;
        } else {
            lo = mid + 1;
        }
    }
    false
}

File: internal/content/testdata/bench/go/perfect-square-check.go
func is_perfect_square(num int) bool {
	if num < 1 {
		return false
	}
	lo, hi := 1, num
	for lo <= hi {
		mid := (lo + hi) / 2
		sq := mid * mid
		if sq == num {
			return true
		} else if sq < num {
			lo = mid + 1
		} else {
			hi = mid - 1
		}
	}
	return false
}
